server/routes/visitJourney/dateAndTime.ts

```diff
@@ -24,65 +24,54 @@ export default class DateAndTime {
     })
 
     let restrictionChangeMessage = ''
-    let matchingSlot = ''
-    let selectedSlot
+    let matchingSlot
 
+    // first time here on update journey, visitSlot.id will be ''
     if (isUpdate && visitSessionData.visitSlot?.id === '') {
-      selectedSlot = getSelectedSlotByStartTimestamp(
+      matchingSlot = getSlotByStartTimeAndRestriction(
         slotsList,
         visitSessionData.visitSlot.startTimestamp,
         visitSessionData.visitRestriction,
       )
 
-      if (selectedSlot) {
-        matchingSlot = selectedSlot.id
+      if (
```

Comment:
Crux of the change here, to be more precise about when a slot will be shown as pre-selected on date/time page.